src/main/java/org/broadinstitute/consent/http/service/DaaService.java

```diff
@@ -102,6 +109,30 @@ public DataAccessAgreement findById(Integer daaId) {
     throw new NotFoundException("Could not find DAA with the provided ID: " + daaId);
   }
 
+  public void sendDaaRequestEmails(User user, Integer daaId) throws Exception {
+    try {
+      List<Institution> institutions = institutionService.findAllInstitutions();
+      Institution institution = institutions.stream()
+            .filter(inst -> inst.getId().equals(user.getInstitutionId()))
+          .findFirst()
+          .orElseThrow();
```

Comment:
If we are adding the institution to the user, do we need to do this additional query/filter? We should have that info handy ... and if not, we can throw an error here.

Reply:
Ok, I think I see where you're coming from. In this method, what I think should happen is the following:
1. Keep the user as an argument
2. Use `InstitutionDAO` instead of institution service to get the full institution. One thing we try to avoid is making services depend on other services. There are exceptions (i.e. `EmailService` and `GCSService`), but if we can, we try. In this case, I think we can avoid the service.
3. `InstitutionDAO` doesn't actually have the thing we want, which is institution with it's SOs. I suggest adding a new DAO method that looks like `findAllInstitutions` but just for a single institution id.
4. With the `SimplifiedUser` entities on your brand new method ☝🏽, you'll be able to avoid the large DB call to find ALL institutions.
5. Then you can send out emails to each simplified user.

Reply:
Untested, but this should be close to what you want:
```
  @RegisterBeanMapper(value = User.class, prefix = "u")
  @RegisterBeanMapper(value = SimplifiedUser.class, prefix = "so")
  @UseRowReducer(InstitutionWithUsersReducer.class)
  @SqlQuery("""
        SELECT i.*,
        u.user_id AS u_user_id, 
        u.email AS u_email,
        u.display_name AS u_display_name, 
        u.create_date AS u_create_date,
        u.email_preference AS u_email_preference,
        u.era_commons_id AS u_era_commons_id,
        u2.user_id AS u2_user_id, u2.email AS u2_email,
        u2.display_name AS u2_display_name, u2.create_date AS u2_create_date,
        u2.email_preference AS u2_email_preference,
        u2.era_commons_id AS u2_era_commons_id,
        so.so_user_id, so.so_email, so.so_display_name
        FROM institution i
        LEFT JOIN users u ON u.user_id = i.create_user
        LEFT JOIN users u2 ON u2.user_id = i.update_user
        LEFT JOIN
          (SELECT
            so.user_id AS so_user_id, so.email AS so_email,
            so.display_name AS so_display_name, so.institution_id AS so_institution_id
            FROM users so
            LEFT JOIN user_role ur ON ur.user_id = so.user_i
            WHERE ur.role_id = 7) so
        ON i.institution_id = so.so_institution_id
        WHERE i.institution_id = :institutionId
      """)
  Institution findInstitutionByIdWithSOs(Integer institutionId);
```